components/push/ffi/src/lib.rs

```diff
@@ -140,14 +140,12 @@ pub extern "C" fn push_update(handle: u64, new_token: FfiStr<'_>, error: &mut Ex
 pub extern "C" fn push_verify_connection(handle: u64, error: &mut ExternError) -> *mut c_char {
     log::debug!("push_verify");
     MANAGER.call_with_result_mut(error, handle, |mgr| -> Result<_> {
-        if let Ok(r) = mgr.verify_connection() {
-            if !r {
-                if let Ok(new_endpoints) = mgr.regenerate_endpoints() {
-                    // use a `match` here to resolve return of <_>
-                    return serde_json::to_string(&new_endpoints).map_err(|e| {
-                        push_errors::ErrorKind::TranscodingError(format!("{:?}", e)).into()
-                    });
-                }
+        if !mgr.verify_connection()? {
+            let new_endpoints = mgr.regenerate_endpoints()?;
+            if new_endpoints.is_empty() {
```

Comment:
Is this correct? The condition looks backwards to me.

Reply:
Ah, good catch! messed it up while doing the clippy fix. Looks like testing caught it too.